cursorless-talon/src/csv_overrides.py

```diff
@@ -60,27 +62,25 @@ def init_csv_and_watch_changes(
         `cursorles-settings` dir
         default_values (dict[str, dict]): The default values for the lists to
         be customized in the given csv
-        extra_ignored_values list[str]: Don't throw an exception if any of
+        extra_ignored_values: Don't throw an exception if any of
         these appear as values; just ignore them and don't add them to any list
         allow_unknown_values bool: If unknown values appear, just put them in the list
         default_list_name Optional[str]: If unknown values are allowed, put any
         unknown values in this list
-        no_update_file Optional[bool]: Set this to `TRUE` to indicate that we should
+        no_update_file: Set this to `TRUE` to indicate that we should
         not update the csv. This is used generally in case there was an issue coming up with the default set of values so we don't want to persist those to disk
         pluralize_lists: Create plural version of given lists
     """
-    if extra_ignored_values is None:
-        extra_ignored_values = []
-
     file_path = get_full_path(filename)
     super_default_values = get_super_values(default_values)
+    assert allow_unknown_values == (default_list_name is not None)
```

Comment:
Turns out we have dependent types here! It would've been too annoying to check both `allow_unknown_values` and `default_list_name` in `update_dicts` to appease the type checker, but I'm also too lazy to update the callers of `init_csv_and_watch_changes` right now.